src/genius/services/http-provider/http-data-provider.service.ts

```diff
@@ -0,0 +1,178 @@
+import {Injectable} from '@angular/core';
+import {HttpClient, HttpHeaders} from '@angular/common/http';
+import {Family} from '../../../model/family';
+import {Person} from '../../../model/person';
+import {Observable} from 'rxjs';
+import {DataProvider} from '../data-provider';
+import {map} from 'rxjs/operators';
+import {environment} from '../../../environments/environment';
+import {FamilyDTO, PersonDTO} from "../dto/dtOs";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class HttpDataProvider extends DataProvider {
+
+  public persons: Array<Person>;
+
+  constructor(private http: HttpClient) {
+    super();
+    this.getPersons().subscribe(httpResponseGetPersons => this.persons = httpResponseGetPersons,
+      (errorHttpResponseGetPersons) => {
+        console.error(`Error status: ${errorHttpResponseGetPersons.error?.status}\n Error message: ${errorHttpResponseGetPersons.error?.message}\n Error path: ${errorHttpResponseGetPersons.error?.path}\n`);
+      });
+  }
+
+  protected mapPerson(obj: any): Person {
+    let person = new Person();
+    person.id = obj.id;
+    person.firstName = (obj.name) ? obj.name.first : null;
+    person.middleName = (obj.name) ? obj.name.middle : null;
+    person.lastName = (obj.name) ? obj.name.last : null;
+    person.sex = (obj.gender) ? obj.gender.toLowerCase() : null;
+    person.familyId = obj.parentFamilyId;
+
+    return person;
+  }
+
+  protected mapFamily(obj: any): Family {
+    let family = new Family();
+    family.children = [];
+    family.id = obj.id;
+    family.mother = this.persons.find(person => obj.wife === person.id)
+    family.father = this.persons.find(person => obj.husband === person.id)
+    if (obj.children && obj.children.length > 0) {
+      obj.children.forEach((childId: number) => {
+        let target = this.persons.find(person => person.id === childId)
+        if (target) family.children.push(target)
+      })
+    }
+    return family;
+  }
+
+  public addNewFamily(family: Family): Observable<Object> {
+    const httpOptions = {
+      headers: new HttpHeaders({
+        'Content-Type': 'application/json',
+        'Accept': 'application/json'
+      })
+    };
+
+    let newFamily: FamilyDTO = new FamilyDTO();
+    if (family.father) {
+      newFamily.husband = family.father.id
+    }
+    if (family.mother) {
+      newFamily.wife = family.mother.id
+    }
+    if (family.children) {
+      newFamily.children = family.children.map(child => child.id);
+    }
+
+    if (family.children && family.children.length > 0) {
+      family.children.forEach(child => newFamily.children.push(child.id))
+    }
+
+    return this.http.post(`${environment.url}/families`, newFamily, httpOptions)
+  }
+
+  public addNewPerson(person: Person): Observable<Object> {
+    const httpOptions = {
```

Comment:
post request doesn't return any data to cleint from server, no need to specify `Accept` header here